March into mapness-The Faceoff

Rex9001 opened the pull request:
## About The Pull Request
This Pull Request adds a new ruin to space called the faceoff. This ruin show cases an unfinished asteriod base in which the syndicate and russians had a meeting. Said meeting was interrupted by a Nanotrasen shock team which was nearly successful in killing all personel. The last standing syndicate survivor managed to detonate a viscerator grenade which killed the remaining PCM soldiers. Now he stands with the last of the viscerators waiting for rescue.....
Notable loot:

- Super sticky tape
- Mini egun
- Chameleon gun
- Pipegun
- Syndicate smokes

SDMM Picture
![Skärmbild 2023-03-10 161758](https://user-images.githubusercontent.com/120136161/224353391-57f510c0-a4d1-47bf-a2c0-d94a7bdf6354.png)

What it looks like in game:
![Skärmbild 2023-03-10 163824](https://user-images.githubusercontent.com/120136161/224358429-67d8c7b3-5cd1-4e6a-82aa-934720ffcd68.png)
![Skärmbild 2023-03-10 163838](https://user-images.githubusercontent.com/120136161/224358453-889c305a-9e0c-4dab-bdc6-cb3b699
![Skärmbild 2023-03-10 163852](https://user-images.githubusercontent.com/120136161/224358470-768d3eca-1d68-4b5f-9fad-068316ec194c.png)
dd0ea.png)

### Mapping March
Ckey to recieve rewards: Rex9001

## Why It's Good For The Game
Adds one more space ruin which increases variation in the pool. This ruin also showcases cooperation between the syndicate and the russians against Nanotrasen, something which I think lacks in game representation and overall it makes for a good addition due to this. The ruin is also not just a loot piniata for powergamers but instead has only a few pretty bad guns and some more fun and flaverful items like super sticky tape and the chameleon gun.
## Changelog
:cl:
add: Adds 2 new iron/dark/airless subtypes, textured and smooth_large
add: Adds the_faceoff.dmm Ruin
/:cl:

Rex9001 committed:
Adds a new ruin, The Faceoff

I think we are missing a ruin involving a big ol fight and this fills that void in our hearts!

This ruin show cases an unfinished asteriod base in which the syndicate and russians had a meeting. Said meeting was interrupted by a Nanotrasen shock team which was nearly successful in killing all personel. The last standing syndicate survivor managed to detonate a viscerator grenade which killed the remaining PCM soldiers.

Notable loot:
Super sticky tape
Mini egun
Chameleon gun
Pipegun
Swat armour (nt troopers)
Syndicate smokes

Rex9001 committed:
Update The_faceoff.dmm

Rex9001 committed:
Maybe fixes the lint error

Double stacked powercables

Jacquerel commented on code/datums/ruins/space.dm:
```diff
@@ -350,3 +350,9 @@
 	name = "Alternate Sciences Research Center"
 	description = "When you're messing with the occult, who knows what you're going to get?"
+
+/datum/map_template/ruin/space/The__faceoff
+	id = "The_faceoff"
```
```suggestion
/datum/map_template/ruin/space/the_faceoff
	id = "the_faceoff"
```

Rex9001 replied:
Should be fixed now

Jacquerel reviewed (changes requested):
Lower case only is our style guide here. Preferably for the filename too.

Rex9001 committed:
Naming conventions

Should now be using lowercase

Jacquerel commented on _maps/RandomRuins/SpaceRuins/The_faceoff.dmm:
```diff
@@ -0,0 +1,2453 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
```
I may have led you into a Git Trap where it has not detected that you changed the casing of this filename 🙂
You may be able to fix this by running:
`git mv _maps/RandomRuins/SpaceRuins/The_faceoff.dmm _maps/RandomRuins/SpaceRuins/the_faceoff.dmm`

Rex9001 replied:
Hmm, the filecase is the_faceoff.dmm when I boot it up in vsc

Jacquerel replied:
On _your computer_ the file is called `the_faceoff.dmm` but in the PR the file is still `The_faceoff.dmm`, because git sometimes is a bit picky about picking up case-only changes to file names. 
This might be fine, I am not sure how case-sensitive it is and the tests seem to pass anyway including loading all ruins, but it would be best to try to fix this.

Rex9001 committed:
Possibly fixes the file naming issue

Addust commented:
there's a syndicate logo turf decal for plastitanium floors iirc, use that please

Cheshify commented:
Looks fantastic, especially for a first time contributor. You do a great job of showing battle damage on the map, and it gives me a good idea of what kind of fight went down to lead to this.

Rex9001 committed:
Changes the icon state of the traitor_rune_unfinished to traitor_rune_done

Does not count for objective since it does not have the traitor_rune_sheen component

Rex9001 committed:
Removes the full red striped decal that was on top of one of the titanium walls

Changes something that looked very very ugly

Mothblocks commented on code/datums/ruins/space.dm:
```diff
@@ -350,3 +350,9 @@
 	name = "Alternate Sciences Research Center"
 	description = "When you're messing with the occult, who knows what you're going to get?"
+
+/datum/map_template/ruin/space/the__faceoff
```
One `_`, not two

Rex9001 replied:
My bad, should be fixed now.

Rex9001 committed:
Fixes formating issue with the datum

Fixes what mothblocks pointed out

MMMiracles commented:
Pretty decent for a first-time contribution, certainly ahead of a lot of our older content. I would recommend watching out using shuttle walls with lights around corners, since those automatically become diagonal in-game (as you can see from your in-game previews). There are wall variants underneath the titanium/plastitanium (nodiagonal) which prevent them from smoothing.
![image](https://user-images.githubusercontent.com/9276171/224460509-1c54cb80-0108-4e5d-9d38-6695e2c3e838.png)
They're distinguished in the map editor by the diagonal lines across them. If you get those cleaned up I don't see many issues with the ruin loot-wise.

Rex9001 committed:
Changes the plastitanium to nodiagonal

Does what mmi asked

Rex9001 commented:
> Pretty decent for a first-time contribution, certainly ahead of a lot of our older content. I would recommend watching out using shuttle walls with lights around corners, since those automatically become diagonal in-game (as you can see from your in-game previews). There are wall variants underneath the titanium/plastitanium (nodiagonal) which prevent them from smoothing. ![image](https://user-images.githubusercontent.com/9276171/224460509-1c54cb80-0108-4e5d-9d38-6695e2c3e838.png) They're distinguished in the map editor by the _cardinal_ lines across them. If you get those cleaned up I don't see many issues with the ruin loot-wise.

Thanks, was not aware there was a non-diagonal variant. I updated all the plastitanium since the titanium walls dont actually have any lights on them

Rex9001 committed:
Adds in more small details

Some bloody tracks from the mining bot and some 40mm ammo to replicate a known pirate weapon

Rex9001 committed:
Moved 3 chairs so it looks a bit more scattered

does what it says on the tin

Rex9001 committed:
Changed storage area

Added a microwave and some donk pockets, to show what the construction crew ate. Even though its so frequent a syndicate toolbox has added to explain how the place was built. The storage area has been slightly increased and it now looks as if  the russians simply sweeped all their loose stuff into the storage area.

Rex9001 committed:
Update the_faceoff.dmm

Rex9001 commented:
Double commit by accident, the second one does the same thing as the changed storage area commit

Rex9001 commented:
Update of what the ruin looks like with the commits:
![Skärmbild 2023-03-13 201316](https://user-images.githubusercontent.com/120136161/224806930-c7e62c31-662a-4a27-8425-aa55e354dafb.png)

Maurukas reviewed (changes requested):
This looks fantastic. This is just the sort of thing I want to see for ruins.

<details>
<summary>You have active turfs</summary>

![image](https://user-images.githubusercontent.com/66576896/224870840-2b278203-a6d9-4f80-b14d-ecc78ba52eae.png)
</details>

This seems to be primarily the `/turf/open/floor/iron/dark/textured` turfs, which does not have an airless subtype. This results in the map loading turfs with vacuum next to turfs with air, and they have to recalculate after the map starts.

There are several ways you can fix this:
* You can create a new `/turf/open/floor/iron/dark/textured/airless` subtype and use that (look at `/turf/open/floor/iron/dark/airless` for how to do it).
* You can use a different airless turf to replace them.

To check for active turfs, get your ruin to spawn in game at roundstart, use the Debug Menu to "Enable Mapping Verbs` . In the mapping menu use "Show Roundstart AT list' and "Show Roundstart AT markers' to debug active turfs.

If you need any help please ask in the mapping event channel or mapping-general.

Mothblocks commented:
Why are active turfs not being caught by CI? :(

Rex9001 committed:
Adds 2  airless tiles and updates the map to use them

Adds 2 new subtypes /turf/open/floor/iron/dark/textured/airless and /turf/open/floor/iron/dark/smooth_large/airless

Updates the map to use these, no at markers now! :pog:

san7890 commented:
you must add this to https://github.com/tgstation/tgstation/blob/master/config/spaceruinblacklist.txt

Rex9001 committed:
Merge branch 'master' into Ruins-map-1

Rex9001 committed:
Adds the_faceoff to the config

Lets you turn off/on my shitty map design

Rex9001 commented:
Added to the spaceruin config I had no idea this needed to be done, my bad

Rex9001 committed:
Merge branch 'master' into Ruins-map-1

san7890 commented:
> Added to the spaceruin config I had no idea this needed to be done, my bad

it's okay a lot of people didn't. i just added it to https://github.com/tgstation/tgstation/blob/master/.github/guides/MAPS_AND_AWAY_MISSIONS.md#configuration so hopefully more people can be more mindful of it or at least help with correcting people whenever they forget in the future.

Rex9001 committed:
Merge branch 'master' into Ruins-map-1

Rex9001 committed:
Merge branch 'master' into Ruins-map-1

Rex9001 committed:
Merge branch 'master' into Ruins-map-1

Rex9001 committed:
Merge branch 'master' into Ruins-map-1

Addust commented:
Critique now that this is merged:

Viscerators are OP, and they can VERY quickly delimb an unwary explorer testing the waters of the ruin and end their round, even if only after a short passby, and are too hard to flee.

No GPS signal.

Rex9001 commented:
> Critique now that this is merged:
> 
> Viscerators are OP, and they can VERY quickly delimb an unwary explorer testing the waters of the ruin and end their round, even if only after a short passby, and are too hard to flee.
> 
> No GPS signal.

I know no GPS singnal kinda sucks but its for flavor of it being a secret base, the viscerators are really strong though yeah. Try to escape them or get in through the many open asteriod sections (just use a pickaxe)
